elastic/kibana
# [Upgrade Assistant] Redesign overview page

This PR implements the first phase of the Upgrade Assistant redesign: a new Overview page and the start to distinguish between Elasticsearch and other stack (i.e., Kibana) deprecations. 

**Other notable changes:**
- Client-side routing
- Breadcrumbs
- Changed the behavior of the "level" filter buttons for ES deprecations. Instead of filtering by "all" or "critical", we now display all deprecations by default and the user can toggle the filters on and off for "critical" and "warning".

**What's next**
- After this PR is merged, I will open up a subsequent PR to incorporate Kibana deprecations (I felt the changes were too large to include here). The UX will be updated to look something like this:

<img width="1260" alt="Screen Shot 2021-04-11 at 7 59 54 PM" src="https://user-images.githubusercontent.com/5226211/114325940-8ba3b880-9b00-11eb-809d-6462378e654f.png">

- I have not changed the UX for the list of ES deprecations or reindexing flow. This will addressed in a separate PR. 
- We've discussed enhancing the overview page by including a "highlights" section that showcases new features in 8.0. This will come as I gain more information on the type of content we wish to show here.

### How to test/review
Note for `7.13`, users will still see the "coming soon" prompt. In order to test these changes, you will need to disable that functionality. In `x-pack/plugins/upgrade_assistant/common/constants.ts`, set `UA_READONLY_MODE` to `false`.

You can add some mock deprecation data by modifying the response returned from `getUpgradeAssistantStatus()` in `x-pack/plugins/upgrade_assistant/server/lib/es_migration_apis.ts`. I've created a gist with some mock data: https://gist.github.com/alisonelizabeth/f093c1464ffd90f9781c42399c758ab4.

- General design feedback (with the understanding that this is the first iteration 😄)
- Verify deprecation logging toggle
- Verify routing/breadcrumbs
- Verify "level" filter functionality on ES deprecations page
- Verify no regressions in ES deprecations list (i.e., search still works, toggle accordions, etc.)

### Screenshots
// Note: This warning callout was pre-existing. It may or may not be applicable anymore depending on which release we decide to re-enable UA. I decided to leave for now.

<img width="1238" alt="Screen Shot 2021-04-11 at 7 56 29 PM" src="https://user-images.githubusercontent.com/5226211/114325958-93fbf380-9b00-11eb-8b66-5650ba4957cb.png">

<img width="1262" alt="Screen Shot 2021-04-08 at 10 46 08 AM" src="https://user-images.githubusercontent.com/5226211/114051072-cab0e000-985a-11eb-8adc-6f3c0ce711bd.png">

<img width="1258" alt="Screen Shot 2021-03-30 at 3 02 45 PM" src="https://user-images.githubusercontent.com/5226211/113043976-2b5d6000-916b-11eb-9ae7-8428d5bc17e3.png">

<img width="1250" alt="Screen Shot 2021-03-30 at 3 05 13 PM" src="https://user-images.githubusercontent.com/5226211/113043979-2bf5f680-916b-11eb-944e-6fba4a18bd02.png">

<img width="1239" alt="Screen Shot 2021-03-30 at 3 08 16 PM" src="https://user-images.githubusercontent.com/5226211/113043980-2c8e8d00-916b-11eb-92ae-ac599812143a.png">

<img width="1255" alt="Screen Shot 2021-03-30 at 3 08 50 PM" src="https://user-images.githubusercontent.com/5226211/113043984-2d272380-916b-11eb-96ac-6e28deeaf3e4.png">

<img width="462" alt="Screen Shot 2021-04-08 at 11 16 20 AM" src="https://user-images.githubusercontent.com/5226211/114052282-e963a680-985b-11eb-8e86-0886544b8bd4.png">

<img width="469" alt="Screen Shot 2021-04-08 at 11 16 26 AM" src="https://user-images.githubusercontent.com/5226211/114052284-e9fc3d00-985b-11eb-8595-8306fa9f74a5.png">

<img width="527" alt="Screen Shot 2021-04-13 at 9 30 09 AM" src="https://user-images.githubusercontent.com/5226211/114560648-e17c7b80-9c3a-11eb-8346-88439b3121f0.png">
